q2_minimap2/tests/test_trim_long_reads.py

```diff
@@ -0,0 +1,249 @@
+            f
+            for f in os.listdir(str(trimmed_se_maxlen_10000))
+            if f.endswith(".fastq.gz")
+        ]
+
+        # Process each FASTQ.GZ file
+        for obs_fp in fastq_files:
+            file_path = os.path.join(str(trimmed_se_maxlen_10000), obs_fp)
+            with gzip.open(file_path, "rt") as obs_fh:
+                self.assertNotEqual(len(obs_fh.readlines()), 0)
+                obs_fh.seek(0)
+                # Iterate over expected and observed reads, side-by-side
+                for records in itertools.zip_longest(*[obs_fh] * 4):
+                    (obs_seq_h, obs_seq, _, obs_qual) = records
+                    # Make sure seqs that do not map to genome were removed
+                    obs_id = obs_seq_h.strip("@\n")
+                    self.assertTrue(obs_id in seq_ids_maxlen10000)
+
+    def test_trimmed_se_minlen_10000(self):
+        query_reads = CasavaOneEightSingleLanePerSampleDirFmt(
+            self.source_dir_se, mode="r"
+        )
+
+        trimmed_se_minlen_10000 = trim(query_reads, minlength=10000)
+        fastq_files = [
+            f
+            for f in os.listdir(str(trimmed_se_minlen_10000))
+            if f.endswith(".fastq.gz")
+        ]
+
+        # Process each FASTQ.GZ file
+        for obs_fp in fastq_files:
+            file_path = os.path.join(str(trimmed_se_minlen_10000), obs_fp)
+            with gzip.open(file_path, "rt") as obs_fh:
+                self.assertNotEqual(len(obs_fh.readlines()), 0)
+                obs_fh.seek(0)
+                # Iterate over expected and observed reads, side-by-side
+                for records in itertools.zip_longest(*[obs_fh] * 4):
+                    (obs_seq_h, obs_seq, _, obs_qual) = records
+                    # Make sure seqs that do not map to genome were removed
+                    obs_id = obs_seq_h.strip("@\n")
+                    self.assertTrue(obs_id in seq_ids_minlen10000)
+
+    def test_trimmed_pe_minq_20(self):
+        # Initialize the query reads from the temporary directory
+        query_reads = CasavaOneEightSingleLanePerSampleDirFmt(
+            self.source_dir_pe, mode="r"
+        )
+        trimmed_pe_minlen_10000 = trim(query_reads, quality=20)
+        fastq_files = [
+            f
+            for f in os.listdir(str(trimmed_pe_minlen_10000))
+            if f.endswith(".fastq.gz")
+        ]
+
+        # Process each FASTQ.GZ file
+        for obs_fp in fastq_files:
+            file_path = os.path.join(str(trimmed_pe_minlen_10000), obs_fp)
+            with gzip.open(file_path, "rt") as obs_fh:
+                # Iterate over expected and observed reads, side-by-side
+                for records in itertools.zip_longest(*[obs_fh] * 4):
+                    (obs_seq_h, obs_seq, _, obs_qual) = records
+                    # Make sure seqs that do not map to genome were removed
+                    obs_id = obs_seq_h.split()[0]
+                    self.assertTrue(obs_id in seq_ids_pe_minq_20)
```

Comment:
```suggestion
        fastq_files = [
            os.path.join(str(trimmed_se_maxlen_10000), obs_fp)
            for f in os.listdir(str(trimmed_se_maxlen_10000))
            if f.endswith(".fastq.gz")
        ]

        # Process each FASTQ.GZ file
        for file_path in fastq_files:
            with gzip.open(file_path, "rt") as obs_fh:
                self.assertNotEqual(len(obs_fh.readlines()), 0)
                obs_fh.seek(0)
                # Iterate over expected and observed reads, side-by-side
                for records in itertools.zip_longest(*[obs_fh] * 4):
                    (obs_seq_h, obs_seq, _, obs_qual) = records
                    # Make sure seqs that do not map to genome were removed
                    obs_id = obs_seq_h.strip("@\n")
                    self.assertTrue(obs_id in seq_ids_maxlen10000)

    def test_trimmed_se_minlen_10000(self):
        query_reads = CasavaOneEightSingleLanePerSampleDirFmt(
            self.source_dir_se, mode="r"
        )

        trimmed_se_minlen_10000 = trim(query_reads, minlength=10000)
        fastq_files = [
            os.path.join(str(trimmed_se_minlen_10000), obs_fp)
            for f in os.listdir(str(trimmed_se_minlen_10000))
            if f.endswith(".fastq.gz")
        ]

        # Process each FASTQ.GZ file
        for file_path in fastq_files:
            with gzip.open(file_path, "rt") as obs_fh:
                self.assertNotEqual(len(obs_fh.readlines()), 0)
                obs_fh.seek(0)
                # Iterate over expected and observed reads, side-by-side
                for records in itertools.zip_longest(*[obs_fh] * 4):
                    (obs_seq_h, obs_seq, _, obs_qual) = records
                    # Make sure seqs that do not map to genome were removed
                    obs_id = obs_seq_h.strip("@\n")
                    self.assertTrue(obs_id in seq_ids_minlen10000)

    def test_trimmed_pe_minq_20(self):
        # Initialize the query reads from the temporary directory
        query_reads = CasavaOneEightSingleLanePerSampleDirFmt(
            self.source_dir_pe, mode="r"
        )
        trimmed_pe_minlen_10000 = trim(query_reads, quality=20)
        fastq_files = [
            os.path.join(str(trimmed_pe_minlen_10000), obs_fp)
            for f in os.listdir(str(trimmed_pe_minlen_10000))
            if f.endswith(".fastq.gz")
        ]

        # Process each FASTQ.GZ file
        for file_path in fastq_files:
            with gzip.open(file_path, "rt") as obs_fh:
                # Iterate over expected and observed reads, side-by-side
                for records in itertools.zip_longest(*[obs_fh] * 4):
                    (obs_seq_h, obs_seq, _, obs_qual) = records
                    # Make sure seqs that do not map to genome were removed
                    obs_id = obs_seq_h.split()[0]
                    self.assertTrue(obs_id in seq_ids_pe_minq_20)
```

Here I just propose that you construct the paths already in the previous step instead of in the loop. Really not important, I just think its a little more readable.

Reply:
Good suggestion! I need to make some corrections as I get "name 'obs_fp' is not defined" with this code.

Reply:
```suggestion
        fastq_files = [
            os.path.join(str(trimmed_se_maxlen_10000), f)
            for f in os.listdir(str(trimmed_se_maxlen_10000))
            if f.endswith(".fastq.gz")
        ]

        # Process each FASTQ.GZ file
        for file_path in fastq_files:
            with gzip.open(file_path, "rt") as obs_fh:
                self.assertNotEqual(len(obs_fh.readlines()), 0)
                obs_fh.seek(0)
                # Iterate over expected and observed reads, side-by-side
                for records in itertools.zip_longest(*[obs_fh] * 4):
                    (obs_seq_h, obs_seq, _, obs_qual) = records
                    # Make sure seqs that do not map to genome were removed
                    obs_id = obs_seq_h.strip("@\n")
                    self.assertTrue(obs_id in seq_ids_maxlen10000)

    def test_trimmed_se_minlen_10000(self):
        query_reads = CasavaOneEightSingleLanePerSampleDirFmt(self.source_dir_se, mode="r")
        trimmed_se_minlen_10000 = trim(query_reads, minlength=10000)
        
        fastq_files = [
            os.path.join(str(trimmed_se_minlen_10000), f)
            for f in os.listdir(str(trimmed_se_minlen_10000))
            if f.endswith(".fastq.gz")
        ]

        # Process each FASTQ.GZ file
        for file_path in fastq_files:
            with gzip.open(file_path, "rt") as obs_fh:
                self.assertNotEqual(len(obs_fh.readlines()), 0)
                obs_fh.seek(0)
                # Iterate over expected and observed reads, side-by-side
                for records in itertools.zip_longest(*[obs_fh] * 4):
                    (obs_seq_h, obs_seq, _, obs_qual) = records
                    # Make sure seqs that do not map to genome were removed
                    obs_id = obs_seq_h.strip("@\n")
                    self.assertTrue(obs_id in seq_ids_minlen10000)

    def test_trimmed_pe_minq_20(self):
        # Initialize the query reads from the temporary directory
        query_reads = CasavaOneEightSingleLanePerSampleDirFmt(self.source_dir_pe, mode="r")
        trimmed_pe_minlen_10000 = trim(query_reads, quality=20)
        
        fastq_files = [
            os.path.join(str(trimmed_pe_minlen_10000), f)
            for f in os.listdir(str(trimmed_pe_minlen_10000))
            if f.endswith(".fastq.gz")
        ]

        # Process each FASTQ.GZ file
        for file_path in fastq_files:
            with gzip.open(file_path, "rt") as obs_fh:
                # Iterate over expected and observed reads, side-by-side
                for records in itertools.zip_longest(*[obs_fh] * 4):
                    (obs_seq_h, obs_seq, _, obs_qual) = records
                    # Make sure seqs that do not map to genome were removed
                    obs_id = obs_seq_h.split()[0]
                    self.assertTrue(obs_id in seq_ids_pe_minq_20)
```